digdag-cli/src/main/java/io/digdag/cli/Sched.java

```diff
@@ -100,56 +102,62 @@ private void startScheduler()
         }
 
+        ConfigElement ce = PropertyUtils.toConfigElement(props);
+        ServerConfig serverConfig = ServerConfig.convertFrom(ce);
+
         // this method doesn't block. it starts some non-daemon threads, setup shutdown handlers, and returns immediately
-        ServerBootstrap.startServer(version, props, SchedulerServerBootStrap.class);
+        ServerBootstrap.start(new SchedulerServerBootStrap(version, serverConfig));
     }
 
     public static class SchedulerServerBootStrap
             extends ServerBootstrap
     {
-        @Inject
-        public SchedulerServerBootStrap(GuiceRsServerControl control)
+        public SchedulerServerBootStrap(Version version, ServerConfig serverConfig)
         {
-            super(control);
+            super(version, serverConfig);
         }
 
         @Override
-        public Injector initialize(ServletContext context)
+        public Bootstrap bootstrap()
         {
-            Injector injector = super.initialize(context);
-
-            ConfigFactory cf = injector.getInstance(ConfigFactory.class);
-            RevisionAutoReloader autoReloader = injector.getInstance(RevisionAutoReloader.class);
-            ConfigLoaderManager loader = injector.getInstance(ConfigLoaderManager.class);
-            ProjectArchiveLoader projectLoader = injector.getInstance(ProjectArchiveLoader.class);
+            return super.bootstrap()
+                .addModules((binder) -> {
+                    binder.bind(RevisionAutoReloader.class).in(Scopes.SINGLETON);
+                    binder.bind(RevisionAutoReloaderStarter.class).asEagerSingleton();
+                })
+                .overrideModulesWith((binder) -> {
+                    // overwrite server that uses ExtractArchiveManager
+                    binder.bind(WorkspaceManager.class).to(LocalWorkspaceManager.class).in(Scopes.SINGLETON);
+                });
+        }
+    }
 
-            Config systemConfig = injector.getInstance(Config.class);
+    private static class RevisionAutoReloaderStarter
+    {
+        private final String projectDirName;
+        private final Config overwriteParams;
+        private final ProjectArchiveLoader projectLoader;
+        private final RevisionAutoReloader autoReloader;
 
-            String projectDirName = systemConfig.get(SYSTEM_CONFIG_AUTO_LOAD_LOCAL_PROJECT_KEY, String.class);
-            Config overwriteParams = cf.fromJsonString(systemConfig.get(SYSTEM_CONFIG_LOCAL_OVERWRITE_PARAMS, String.class));
+        @Inject
+        public RevisionAutoReloaderStarter(Config systemConfig, ProjectArchiveLoader projectLoader, ConfigFactory cf, RevisionAutoReloader autoReloader)
+        {
+            this.projectDirName = systemConfig.get(SYSTEM_CONFIG_AUTO_LOAD_LOCAL_PROJECT_KEY, String.class);
+            this.overwriteParams = cf.fromJsonString(systemConfig.get(SYSTEM_CONFIG_LOCAL_OVERWRITE_PARAMS, String.class));
```

Comment:
oh ok...`overwriteParams` is used a lot in digdag's code (especially digdag-cli but also some in digdag-core such as ProjectArchiveLoader).
should all of them be replaced?